### PR TITLE
Clarify how additive changes are handled

### PR DESCRIPTION
**Pull request by tigrannajaryan**
This PR explicitly lists the additive changes allowed and adds a requirement that such additive changes must be accompanied by interoperability explanation when necessary.

This is a subset of https://github.com/open-telemetry/opentelemetry-proto/pull/432 that contains other guarantees that we did not yet agree to. I believe this particular subset is necessary regardless of what we decide about #432.

### CODE REVIEW

**Review by jack-berg — APPROVED** [review]

**Review by pellared — APPROVED** [review]
I think we can use https://github.com/bufbuild/buf (see: [`buf breaking`](https://buf.build/docs/breaking/usage)) to automate this rule.

**Review by carlosalberto — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
@open-telemetry/specs-approvers this has enough approvals. Please speak if you have any objections otherwise I will merge.

**Comment by tigrannajaryan** [comment]
@open-telemetry/specs-approvers please review.

**Pull request merged**
